src/main/java/network/brightspots/rcv/GuiConfigController.java

```diff
@@ -796,18 +817,27 @@ public LocalDate fromString(String string) {
     choiceCvrProvider.setOnAction(event -> {
       clearAndDisableCvrFilesTabFields();
       String provider = getChoiceElse(choiceCvrProvider, Provider.PROVIDER_UNKNOWN);
-      if (provider.equals("ES&S")) {
-        buttonAddCvrFile.setDisable(false);
-        textFieldCvrFilePath.setDisable(false);
-        buttonCvrFilePath.setDisable(false);
-        textFieldCvrFirstVoteCol.setDisable(false);
-        textFieldCvrFirstVoteRow.setDisable(false);
-        textFieldCvrIdCol.setDisable(false);
-        textFieldCvrPrecinctCol.setDisable(false);
-      } else if (provider.equals("CDF") || provider.equals("Dominion") || provider.equals("Hart")) {
-        buttonAddCvrFile.setDisable(false);
-        textFieldCvrFilePath.setDisable(false);
-        buttonCvrFilePath.setDisable(false);
+      switch (provider) {
+        case "ES&S" -> {
+          buttonAddCvrFile.setDisable(false);
+          textFieldCvrFilePath.setDisable(false);
+          buttonCvrFilePath.setDisable(false);
+          textFieldCvrFirstVoteCol.setDisable(false);
+          textFieldCvrFirstVoteRow.setDisable(false);
+          textFieldCvrIdCol.setDisable(false);
+          textFieldCvrPrecinctCol.setDisable(false);
+        }
+        case "CDF" -> {
+          buttonAddCvrFile.setDisable(false);
+          textFieldCvrFilePath.setDisable(false);
+          buttonCvrFilePath.setDisable(false);
```

Comment:
These three are common to all three cases, so you could just do them outside of the switch.

Reply:
True, I thought about this, but kept it because it causes an unimplemented provider to not affect the GUI, which seems desirable. Plus it forces us to explicitly lay out what we expect to happen for each provider.